src/test/ui/regions/regions-proc-bound-capture.stderr

```diff
@@ -1,12 +1,20 @@
-error[E0621]: explicit lifetime required in the type of `x`
-  --> $DIR/regions-proc-bound-capture.rs:9:5
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/regions-proc-bound-capture.rs:9:14
    |
-LL | fn static_proc(x: &isize) -> Box<dyn FnMut()->(isize) + 'static> {
-   |                   ------ help: add explicit lifetime `'static` to the type of `x`: `&'static isize`
+LL | fn static_proc(x: &isize) -> Box<dyn FnMut() -> (isize) + 'static> {
+   |                   ------ this data with an anonymous lifetime `'_`...
 LL |     // This is illegal, because the region bound on `proc` is 'static.
-LL |     Box::new(move|| { *x })
-   |     ^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'static` required
+LL |     Box::new(move || { *x })
+   |              ^^^^^^^^^^^^^^ ...is captured here requiring it to live as long as `'static`
+   |
+help: consider changing the trait object's explicit `'static` bound to an anonymous lifetime `'_`
```

Comment:
another example. It feels like it's not just "an anonymous lifetime", it's *the lifetime from the argument `x`*. 

help: consider changing the trait object's explicit `'static` bound to `'_`, to declare that you capture data from the argument `x`

Reply:
Current output:

<img width="713" alt="Screen Shot 2020-06-02 at 4 06 32 PM" src="https://user-images.githubusercontent.com/1606434/83578219-0cf6eb80-a4eb-11ea-966a-56f1e349685f.png">

Reply:
Nice!